src/execution/command_parser.c

```diff
@@ -115,22 +190,26 @@ static int _ModelRunCtx_SetParams(RedisModuleCtx *ctx, RedisModuleString **inkey
     for (size_t i = 0; i < noutputs; i++) {
         if (model->outputs)
             opname = model->outputs[i];
+        _AnalyzeKey(outkeys[i], err); // Relevant for enterprise cluster.
```

Comment:
do it on parse time

Reply:
We can't because the parsing function is used from dag command as well (and in DAG we do not take the keys from key space...)
This is called only from "pure" AI.MODELEXECUTE.